apps/app.go

```diff
@@ -110,20 +64,50 @@ type Manifest struct {
 	// "/command/apptrigger"}``.
 	RequestedLocations Locations `json:"requested_locations,omitempty"`
 
-	// By default invoke "/install", expanding App, AdminAccessToken, and
-	// Config.
-	OnInstall   *Call `json:"on_install,omitempty"`
-	OnUninstall *Call `json:"on_uninstall,omitempty"`
-	OnStartup   *Call `json:"on_startup,omitempty"`
+	// Functions must be included by the developer in the published manifest for
+	// AWS apps. These declarations are used to:
+	// - create AWS Lambda functions that will service requests in Mattermost
+	// Cloud;
+	// - define path->function mappings, aka "routes". The function with the
+	// path matching as the longest prefix is used to handle a Call request.
+	Functions []Function
+}
 
-	// By default invoke "/bindings".
-	Bindings *Call `json:"bindings,omitempty"`
+// App describes an App installed on a Mattermost instance. App should be
+// abbreviated as `app`.
+type App struct {
+	Manifest
 
-	// Deployment manifest for hostable apps will include path->invoke mappings
-	Functions []Function
+	Disabled bool `json:"disabled,omitempty"`
+
+	// Secret is used to issue JWT
+	Secret string `json:"secret,omitempty"`
+
+	OAuth2ClientID     string `json:"oauth2_client_id,omitempty"`
+	OAuth2ClientSecret string `json:"oauth2_client_secret,omitempty"`
+	OAuth2TrustedApp   bool   `json:"oauth2_trusted_app,omitempty"`
+
+	BotUserID      string `json:"bot_user_id,omitempty"`
+	BotUsername    string `json:"bot_username,omitempty"`
+	BotAccessToken string `json:"bot_access_token,omitempty"`
+
+	// Grants should be scopable in the future, per team, channel, post with
+	// regexp.
+	GrantedPermissions Permissions `json:"granted_permissions,omitempty"`
```

Comment:
Would be useful to document the initial set (AsBot, etc.) like present+future.

Reply:
Doing it in the upcoming Docs PR